code_challenges/code_wars/trigrams.py

```diff
@@ -0,0 +1,17 @@
+def trigrams(phrase):
+    j = 3
+    trigrams = []
+
+    updated_phrase = phrase.replace(" ", "_")
+    
+    for i in range(len(updated_phrase)):
+        if len(updated_phrase[i:j]) == 3:
+            trigrams.append(updated_phrase[i:j])
+            j += 1
+        else:
+            return " ".join(trigrams)
```

Comment:
Now that you are iterating on a range, how about not including the full length since you want to stop `j` short of the end?  Then you wont need the if check.

Something like:
```python
def trigrams(phrase):
    chars_per_ngram = 3
    ngrams = []
    phrase = phrase.replace(" ", "_")
    for i in range(len(phrase) - chars_per_ngram + 1):
        ngrams.append(phrase[i:i + chars_per_ngram])
    return " ".join(ngrams)
```

Reply:
i was taking 2 off the end but range(len(phrase) - Chars_per_ngram + 1) is stopping before i get to the last 2